File: paykit-ffi/src/swift/AllowanceBindingsCompile.swift
// Compile-only consumer fixture for the generated Allowance API.
private func compileAllowanceBindingsSurface(
    sdk: PaykitSdkProtocol,
    counterparty: String,
    receiverPath: String,
    allowanceId: String
) async throws {
    let amountRange = try AllowanceAmountRange(minimum: "1", maximum: "10")
    let period = try AllowancePeriod(kind: "rolling", every: 1, unit: "day", anchor: nil)
    let periodLimit = try AllowancePeriodLimit(
        amountLimit: "25",
        paymentCountLimit: 2,
        period: period
    )
    let terms = try AllowanceTerms(
        asset: "USD",
        perPaymentAmount: amountRange,
        periodLimits: [periodLimit],
        lifetimeAmountLimit: nil,
        activeFrom: nil,
        expiresAt: nil,
        allowedPaymentEndpointIdentifiers: nil
    )
    let amountRangeProtocol: AllowanceAmountRangeProtocol = amountRange
    let periodProtocol: AllowancePeriodProtocol = period
    let periodLimitProtocol: AllowancePeriodLimitProtocol = periodLimit
    let termsProtocol: AllowanceTermsProtocol = terms
    let filter = AllowanceFilter(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        localRole: .allower,
        states: [.proposed]
    )
    let historyStatus: AllowanceHistoryStatus = .consistent

    let listed: [AllowanceRecord] = try await sdk.listAllowances(filter: filter)
    let found: AllowanceRecord? = try await sdk.getAllowance(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        allowanceId: allowanceId
    )
    let proposed: AllowanceRecord = try await sdk.proposeAllowance(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        localRole: .allowee,
        terms: terms
    )
    let accepted: AllowanceRecord = try await sdk.acceptAllowance(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        allowanceId: allowanceId
    )
    let rejected: AllowanceRecord = try await sdk.rejectAllowance(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        allowanceId: allowanceId
    )
    let ended: AllowanceRecord = try await sdk.endAllowance(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        allowanceId: allowanceId
    )

    let proof = try PrivateJsonObject(text: "{}")
    let automaticProof = PaymentProofSubmission(
        billingPeriod: nil,
        paymentEndpointIdentifier: "btc-lightning-bolt11",
        allowanceId: allowanceId,
        proof: proof
    )
    let manualProof = PaymentProofSubmission(
        billingPeriod: nil,
        paymentEndpointIdentifier: "btc-lightning-bolt11",
        allowanceId: nil,
        proof: proof
    )
    let requestWithProof = try await sdk.submitPaymentProof(
        counterparty: counterparty,
        counterpartyReceiverPath: receiverPath,
        paymentRequestId: "550e8400-e29b-41d4-a716-446655440000",
        proof: automaticProof
    )
    let reportedAllowances: [String?] = requestWithProof.paymentProofs.map { $0.allowanceId }

    let privateGetters = (
        amountRange.minimum(),
        period.kind(),
        periodLimit.period(),
        terms.perPaymentAmount()
    )
    let redactedDescriptions = [
        amountRange.description,
        amountRange.debugDescription,
        period.description,
        period.debugDescription,
        periodLimit.description,
        periodLimit.debugDescription,
        terms.description,
        terms.debugDescription,
    ]

    _ = (
        amountRangeProtocol,
        periodProtocol,
        periodLimitProtocol,
        termsProtocol,
        privateGetters,
        redactedDescriptions,
        historyStatus,
        listed,
        found,
        proposed,
        accepted,
        rejected,
        ended,
        manualProof,
        reportedAllowances
    )
}

File: paykit-ffi/bindings/android/lib/src/test/kotlin/com/synonym/paykit/AllowanceBindingsCompile.kt
package com.synonym.paykit

// Compile-only consumer fixture for the generated Allowance API.
@Suppress("UNUSED_VARIABLE")
internal suspend fun compileAllowanceBindingsSurface(
    sdk: PaykitSdkInterface,
    counterparty: String,
    receiverPath: String,
    allowanceId: String,
): Pair<AllowanceTermsInterface, AllowanceHistoryStatus> {
    val amountRange = AllowanceAmountRange(minimum = "1", maximum = "10")
    val period = AllowancePeriod(kind = "rolling", every = 1UL, unit = "day", anchor = null)
    val periodLimit = AllowancePeriodLimit(
        amountLimit = "25",
        paymentCountLimit = 2UL,
        period = period,
    )
    val terms = AllowanceTerms(
        asset = "USD",
        perPaymentAmount = amountRange,
        periodLimits = listOf(periodLimit),
        lifetimeAmountLimit = null,
        activeFrom = null,
        expiresAt = null,
        allowedPaymentEndpointIdentifiers = null,
    )
    val filter = AllowanceFilter(
        counterparty = counterparty,
        counterpartyReceiverPath = receiverPath,
        localRole = AllowanceLocalRole.ALLOWER,
        states = listOf(AllowanceLifecycleState.PROPOSED),
    )

    val listed: List<AllowanceRecord> = sdk.listAllowances(filter)
    val found: AllowanceRecord? = sdk.getAllowance(counterparty, receiverPath, allowanceId)
    val proposed: AllowanceRecord = sdk.proposeAllowance(
        counterparty,
        receiverPath,
        AllowanceLocalRole.ALLOWEE,
        terms,
    )
    val accepted: AllowanceRecord = sdk.acceptAllowance(counterparty, receiverPath, allowanceId)
    val rejected: AllowanceRecord = sdk.rejectAllowance(counterparty, receiverPath, allowanceId)
    val ended: AllowanceRecord = sdk.endAllowance(counterparty, receiverPath, allowanceId)
    val proof = PrivateJsonObject(text = "{}")
    val automaticProof = PaymentProofSubmission(
        billingPeriod = null,
        paymentEndpointIdentifier = "btc-lightning-bolt11",
        allowanceId = allowanceId,
        proof = proof,
    )
    val manualProof = PaymentProofSubmission(
        billingPeriod = null,
        paymentEndpointIdentifier = "btc-lightning-bolt11",
        allowanceId = null,
        proof = proof,
    )
    val requestWithProof = sdk.submitPaymentProof(
        counterparty,
        receiverPath,
        "550e8400-e29b-41d4-a716-446655440000",
        automaticProof,
    )
    val reportedAllowances: List<String?> = requestWithProof.paymentProofs.map { it.allowanceId }
    val amountRangeInterface: AllowanceAmountRangeInterface = amountRange
    val periodInterface: AllowancePeriodInterface = period
    val periodLimitInterface: AllowancePeriodLimitInterface = periodLimit
    val privateGetters = listOf(
        amountRange.minimum(),
        period.kind(),
        periodLimit.amountLimit(),
        terms.asset(),
    )

    return terms to AllowanceHistoryStatus.CONSISTENT
}
